### ISSUE TITLE
Developing informative priors on tree age: missing outer rings

### ISSUE THREAD

**Issue by ebuhle** [kzaret/RQ1v2_PIUVestab#6]
We discussed with Andres the possibility of putting some sort of default "best guess" prior on the number of missing outer rings where applicable. Presumably this would be based on information about weathering rates, etc., although this is still a bit murky to me since it seems like you would also need to know when the tree died. IIRC Andres mentioned some more specific data (?) indicating, encouragingly, that the number of missing outer rings was likely small, but I don't remember what it was.

As this is the most subjective component of the tree age prior, it is also the most reliant on domain knowledge / expert judgment, and probably warrants prior sensitivity checking before final publication. For now, though, it would just be good to have something more plausible than a point-mass prior at zero.

**Comment by kzaret** [comment]
>  Presumably this would be based on information about weathering rates, etc., although this is still a bit murky to me since it seems like you would also need to know when the tree died. IIRC Andres mentioned some more specific data (?) indicating, encouragingly, that the number of missing outer rings was likely small, but I don't remember what it was.

Regarding tree cores from the Burned Forest patch, we have the fire scar data that Andres worked up as part of his dissertation, which will give us potential death dates, and we have his tree cores from live unburned trees nearby the site.  If the tree cores that are missing rings that I collected came from trees similar in diameter (and perhaps minimum age) classes to Andres' cores, we could use his cores as an indication of the missing lengths and thus number of missing rings.  Andres suggested, anecdotally, that ten to fifteen rings missing due to weathering may be reasonable to expect from trees killed by the last fire.  Unfortunately, Andres hasn't yet been able to find his tree core data; I'm about to check the Appendices of his Diss, just in case relevant info. is in there.

For cores missing outermost rings from the other patches, it would also be possible to compare the size classes of the trees from which they were extracted to trees from which complete cores were extracted (except in the Fores Patch where we always missed the pith).  Also, there is the issue that cores were often extracted at varying diameters and diameter estimates are very uncertain for the Forest Patch.

I also want us to take note that while "weathering" may be associated with a certain rate of ring decay, some cores are missing outermost rings because pieces of the core were lost during collection, transportation or processing.

**Comment by kzaret** [comment]
I did a round of QAQC of the cores in PIUV_CoredProcessed.csv in which I examined the scans and/or physical cores that were missing outer rings and, 1) added an "ORW" column indicating whether outer rings were lost due to weathering (outer rings weathered = 1, outer rings lost due to some other means, e.g. the core broke = 0, if a core had outer rings [Outer rings = 1], then ORW = NA [but maybe these should be 0s?]; 2) added a "remove" (i.e., completely remove from analysis) column for cores that were selectively rather than randomly sampled or that were so badly damaged or glued incorrectly that I am not comfortable even thinking of them as "minimum age cores"; and 3) added a "min_age" column indicating the cores/trees for which it would probably be inappropriate to try to estimate the establishment year (e.g., cores in which the bark side was charred rather than simply weathered, so who knows how many rings burned away, or if there was damage/missing rings in the interior of the core).

Here's what a weathered core looks like, in case you're curious:
![DSC08837](https://user-images.githubusercontent.com/73861013/111561182-44204b80-8751-11eb-81db-839c0758aff3.JPG)

**Comment by ebuhle** [comment]
> Regarding tree cores from the Burned Forest patch, we have the fire scar data that Andres worked up as part of his dissertation, which will give us potential death dates, and we have his tree cores from live unburned trees nearby the site. If the tree cores that are missing rings that I collected came from trees similar in diameter (and perhaps minimum age) classes to Andres' cores, we could use his cores as an indication of the missing lengths and thus number of missing rings.

> For cores missing outermost rings from the other patches, it would also be possible to compare the size classes of the trees from which they were extracted to trees from which complete cores were extracted (except in the Fores Patch where we always missed the pith). Also, there is the issue that cores were often extracted at varying diameters and diameter estimates are very uncertain for the Forest Patch.

Hmm, interesting. So it sounds like this might be amenable to modeling after all, but I'm having a hard time wrapping my head around what that model would look like, exactly. Something like `core_length ~ tree_diameter*patch`, maybe also with plot-level coefficients?? Where you would fit the intact-core data from proxy trees (Andres's unburned sample from Burned Forest, or intact cores from the other patches), use the model to predict the length of weathered cores, and then use "average" outer ring width in a Duncan-esque manner to convert missing length to missing rings...??? I'm using intentionally made-up variable names b/c I'm not certain which columns in `PIUV_CoredProcessed.csv` correspond to what. Come to think of it, I'm not even certain whether the core goes all the way through the tree (such that `core_length` and `tree_diameter` are the same thing) or stops at the presumed pith (such that the core length is roughly the radius). I'm also not sure how the death dates in Burned Forest come into this.

Is this approximately what you were driving at, or am I way off?

> I did a round of QAQC of the cores in PIUV_CoredProcessed.csv in which I examined the scans and/or physical cores that were missing outer rings

Thanks for doing this; sounds dangerously close to actual science. So the gray is just oxidation, not some residue of cambium [I just looked that up] or the like?

> 3) added a "min_age" column indicating the cores/trees for which it would probably be inappropriate to try to estimate the establishment year (e.g., cores in which the bark side was charred rather than simply weathered, so who knows how many rings burned away, or if there was damage/missing rings in the interior of the core).

Oof. So `cores` now contains 24/168 rows with `min_age == 1`, including 14/39 from Burned Forest. To discard those from the analysis would be...not good. To get a sense of where I'm coming from here, as well as a sneak preview of the end product of this highly problematized dendrochronological footnote, you can run `total_age_priors.R`. (Note that you need saved .RData from `CorHeightAddRings.R` and `Duncan_rings-to-pith.R`.) That said, this issue is not specific to our fancy analysis method -- you're not getting much blood from the stone if over a third of the data from one site are unusable, and the rest are spread out over 500 years. :confused:

**Comment by kzaret** [comment]
> Hmm, interesting. So it sounds like this might be amenable to modeling after all, but I'm having a hard time wrapping my head around what that model would look like, exactly. Something like core_length ~ tree_diameter*patch, maybe also with plot-level coefficients?? Where you would fit the intact-core data from proxy trees (Andres's unburned sample from Burned Forest, or intact cores from the other patches), use the model to predict the length of weathered cores, and then use "average" outer ring width in a Duncan-esque manner to convert missing length to missing rings...??? I'm using intentionally made-up variable names b/c I'm not certain which columns in PIUV_CoredProcessed.csv correspond to what.

>Is this approximately what you were driving at, or am I way off?

Thanks for thinking through this.  I think that relating core length to number of rings would also somehow need to take coring height into consideration.  However, I doubt that Andres has enough complete samples (cores or cross sections) from the LC site to build a model of core length.  I know of 7 cross sections (and there should be one core associated with each) collected to examine fire history (so these would have been collected from burned trees that may or may not have been missing outer rings on the side of the tree opposite of the fire scar), and his dissertation mentions another subset of cores associated with patch age structure but I can't seem to find a number of samples per site.  All of this may be moot anyway since he hasn't been able to find his raw data (but see below).

> Come to think of it, I'm not even certain whether the core goes all the way through the tree (such that core_length and tree_diameter are the same thing) or stops at the presumed pith (such that the core length is roughly the radius). I'm also not sure how the death dates in Burned Forest come into this.

Yeah, core length is roughly the radius; more 'roughly'  for cores that didn't reach the pith.  And actually, when I measure radii of cores and compare those to the Cor_Diam_cm in PIUV_CoredProcessed.csv, there are discrepancies of over a centimeter or more.  In part, this is because the diameter measurement includes bark on both sides of the tree -- we used DBH tape to make this measurement before extracting the core -- and not all of the bark (from the point at which the increment core was inserted in the tree) was collected and mounted with the ring-bearing part of the core.  This suggests that I would probably need Andres' physical cores to measure their lengths rather than his DBH measurements. . . .But I'm not feeling enthusiastic about this approach to getting at missing rings.

We could return to the idea of cross-dating.  Presumably, I could know the year of formation of the most recent ring.  Andres' fire data would indicate the most likely year of death and his mortality data could give an indication of time lags between fire date and date of death/end of ring growth.  Rather than a modeling exercise, this could be a qualitative way of estimating the number of missing rings.

**Comment by ebuhle** [comment]
> We could return to the idea of cross-dating. Presumably, I could know the year of formation of the most recent ring. Andres' fire data would indicate the most likely year of death and his mortality data could give an indication of time lags between fire date and date of death/end of ring growth. Rather than a modeling exercise, this could be a qualitative way of estimating the number of missing rings.

Oooh, I like this! Even though / especially because it involves you doing all the real work. I appreciate that cross-dating is nontrivial, though, and would be open to something more kludgy as a stopgap -- especially for AGU purposes. Or maybe that won't be necessary? We should talk about that talk sometime, either here (in a dedicated issue?) or offline, as you wish.

**Comment by kzaret** [comment]
>Oooh, I like this! Even though / especially because it involves you doing all the real work. I appreciate that cross-dating is nontrivial, though, and would be open to something more kludgy as a stopgap -- especially for AGU purposes. Or maybe that won't be necessary? We should talk about that talk sometime, either here (in a dedicated issue?) or offline, as you wish.

Yeah, I won't be able to get the crossdating done in time for whatever it reveals to make it into the AAG (American Association of Geographers) talk.  I'm not yet sure about the need for stopgap measures, though I'm about to put some thought into talk.  I've started a new issue and perhaps talk offline as well -- thanks!

**Comment by ebuhle** [comment]
Actually, on further consideration: if you're doing crossdating, couldn't you just use it in the usual way to get the age of the *oldest* ring (or close enough to use Duncan, in the event that the core missed the pith)? At that point the missing outer rings would be irrelevant, no?

**Comment by kzaret** [comment]
Yes, indeed.  Good point.  Maybe I should crossdate a subset of the cores that are missing outer rings but that I believe to be weathered to check the validity of assuming they're missing only a certain range of outer rings.  I don't think that I have the determination or stamina to crossdate all of the cores from the Burned Forest patch, but perhaps that process will go more smoothly than I anticipate?!

Thinking of the AAG talk and a potential publication -- I guess it's naive to think that we could complete a dendro-based study without doing any crossdating. . . .

**Comment by ebuhle** [comment]
>  I don't think that I have the determination or stamina to crossdate all of the cores from the Burned Forest patch, but perhaps that process will go more smoothly than I anticipate?!

Don't you have undergrad slaves for this sort of thing? Would be a bit tragic to get partway to a real solution for this (a complete solution, for the crossdated subset of cores) only to have to kludge it the last mile.
